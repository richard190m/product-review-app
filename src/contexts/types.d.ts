import { ReactNode } from 'react';

export type ReviewProviderProps = {
  children: ReactNode;
};
