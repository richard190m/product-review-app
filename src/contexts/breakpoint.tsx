import { BreakpointContextInterface, BreakpointProviderProps } from './types';
import { createContext, useEffect, useState } from 'react';

export const BreakpointContext =
  createContext<BreakpointContextInterface | null>(null);

export const BreakpointProvider = ({ children }: BreakpointProviderProps) => {
  const [breakpoint, setBreakpoint] = useState<string>('');

  const handleBreakpoint = () => {
    if (window.innerWidth < 768) breakpoint != 'xs' && setBreakpoint('xs');
    else if (window.innerWidth < 1024)
      breakpoint != 'sm' && setBreakpoint('sm');
    else if (window.innerWidth < 1440)
      breakpoint != 'md' && setBreakpoint('md');
    else breakpoint != 'lg' && setBreakpoint('lg');
  };

  useEffect(() => {
    handleBreakpoint();

    function handleResize() {
      handleBreakpoint();
    }

    window.addEventListener('resize', handleResize);

    return () => window.removeEventListener('resize', handleResize);
  }, []);

  return (
    <BreakpointContext.Provider value={{ breakpoint }}>
      {children}
    </BreakpointContext.Provider>
  );
};
