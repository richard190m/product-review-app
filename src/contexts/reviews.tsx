import { createContext, useEffect, useState } from 'react';
import {
  HandleSubmit,
  Review,
  ReviewProviderProps,
  ReviewsContextInterface,
} from './types';
import MockReviews from '../mocks/MockReviews';

export const ReviewsContext = createContext<ReviewsContextInterface | null>(
  null,
);

export const ReviewsProvider = ({ children }: ReviewProviderProps) => {
  const [reviews, setReviews] = useState<Review[]>([]);

  const handleSubmit: HandleSubmit = review => {
    setReviews([...reviews, review]);
  };

  useEffect(() => {
    // Load Reviews (mocks)
    setReviews(MockReviews);
  }, []);

  return (
    <ReviewsContext.Provider value={{ reviews, submit: handleSubmit }}>
      {children}
    </ReviewsContext.Provider>
  );
};
