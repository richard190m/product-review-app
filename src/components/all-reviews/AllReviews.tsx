import {
  AllContainer,
  Reviews,
  ReviewsContainer,
  TrendsStyled,
} from './styles';
import React, { useContext } from 'react';
import Review from './Review/Review';
import { ReviewsContext } from '../../contexts/reviews';
import sortReviewsByDate from '../../utils/sortReviewsByDate';

const AllReviews = () => {
  const reviewsContext = useContext(ReviewsContext);
  sortReviewsByDate(reviewsContext?.reviews);

  return (
    <AllContainer>
      <Reviews>
        <h2>Reviews</h2>
        <ReviewsContainer>
          {reviewsContext?.reviews.map((review, index) => (
            <Review data={review} key={index} />
          ))}
        </ReviewsContainer>
      </Reviews>
      <TrendsStyled />
    </AllContainer>
  );
};

export default AllReviews;
