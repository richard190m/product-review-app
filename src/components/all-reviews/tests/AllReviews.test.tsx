import AllReviews from '../AllReviews';
import { render } from '@testing-library/react';
import { ReviewsProvider } from '../../../contexts/reviews';

jest.mock('../../trends', () => () => <div>Trends</div>);

describe('AllReviews', () => {
  test('should render correctly', () => {
    const { container } = render(
      <ReviewsProvider>
        <AllReviews />
      </ReviewsProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
