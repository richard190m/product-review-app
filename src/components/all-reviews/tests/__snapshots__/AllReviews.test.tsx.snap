// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllReviews should render correctly 1`] = `
<div>
  <section
    class="sc-bczRLJ eemIpn"
  >
    <div
      class="sc-gsnTZi cFgCqY"
    >
      <h2>
        Reviews
      </h2>
      <div
        class="sc-hKMtZM bKMAaB"
      >
        <div
          class="sc-eCYdqJ iyiAJB"
        >
          <div
            class="sc-jSMfEi cxMpge"
          >
            <svg
              aria-hidden="true"
              class="StyledIconBase-ea9ulj-0 lbJwfL sc-ftvSup fLWAMn"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm0 18c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8z"
              />
              <circle
                cx="8.5"
                cy="10.5"
                r="1.5"
              />
              <circle
                cx="15.493"
                cy="10.493"
                r="1.493"
              />
              <path
                d="M12 18c4 0 5-4 5-4H7s1 4 5 4z"
              />
            </svg>
            <p>
              Name 3
            </p>
          </div>
          <div
            class="sc-papXJ gDpZmt"
          >
            <p
              class="sc-breuTD iiijzD"
            >
              <svg
                aria-hidden="true"
                class="StyledIconBase-ea9ulj-0 lbJwfL sc-jqUVSM gKbZDu"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="StyledIconBase-ea9ulj-0 lbJwfL sc-jqUVSM gKbZDu"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="StyledIconBase-ea9ulj-0 lbJwfL sc-jqUVSM gKbZDu"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="StyledIconBase-ea9ulj-0 lbJwfL sc-jqUVSM gKbZDu"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="StyledIconBase-ea9ulj-0 lbJwfL sc-jqUVSM gKbZDu"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
                />
              </svg>
              <span
                class="sc-evZas ccJzNt"
              >
                (01/05/2022)
              </span>
            </p>
            <div
              class="sc-iqcoie gAUmKf"
            >
              <p>
                Good product
              </p>
              <p
                class="sc-crXcEl gQMZmj"
              >
                Best product I ever used
              </p>
            </div>
          </div>
        </div>
        <div
          class="sc-eCYdqJ iyiAJB"
        >
          <div
            class="sc-jSMfEi cxMpge"
          >
            <svg
              aria-hidden="true"
              class="StyledIconBase-ea9ulj-0 lbJwfL sc-iBkjds flrtOA"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm0 18c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8z"
              />
              <circle
                cx="8.5"
                cy="10.5"
                r="1.5"
              />
              <circle
                cx="15.493"
                cy="10.493"
                r="1.493"
              />
              <path
                d="M7.974 15H16v2H7.974z"
              />
            </svg>
            <p>
              Name 2
            </p>
          </div>
          <div
            class="sc-papXJ gDpZmt"
          >
            <p
              class="sc-breuTD iiijzD"
            >
              <svg
                aria-hidden="true"
                class="StyledIconBase-ea9ulj-0 lbJwfL sc-jqUVSM gKbZDu"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="StyledIconBase-ea9ulj-0 lbJwfL sc-jqUVSM gKbZDu"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="StyledIconBase-ea9ulj-0 lbJwfL sc-jqUVSM gKbZDu"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="StyledIconBase-ea9ulj-0 lbJwfL sc-kDDrLX ghEnkw"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="StyledIconBase-ea9ulj-0 lbJwfL sc-kDDrLX ghEnkw"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
                />
              </svg>
              <span
                class="sc-evZas ccJzNt"
              >
                (01/04/2022)
              </span>
            </p>
            <div
              class="sc-iqcoie gAUmKf"
            >
              <p>
                Not bad, not good
              </p>
              <p
                class="sc-crXcEl gQMZmj"
              >
                It's a product that can do the job.
              </p>
            </div>
          </div>
        </div>
        <div
          class="sc-eCYdqJ iyiAJB"
        >
          <div
            class="sc-jSMfEi cxMpge"
          >
            <svg
              aria-hidden="true"
              class="StyledIconBase-ea9ulj-0 lbJwfL sc-gKXOVf cKhAZa"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm0 18c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8z"
              />
              <circle
                cx="8.5"
                cy="10.5"
                r="1.5"
              />
              <circle
                cx="15.493"
                cy="10.493"
                r="1.493"
              />
              <path
                d="M12 14c-3 0-4 3-4 3h8s-1-3-4-3z"
              />
            </svg>
            <p>
              Name 1
            </p>
          </div>
          <div
            class="sc-papXJ gDpZmt"
          >
            <p
              class="sc-breuTD iiijzD"
            >
              <svg
                aria-hidden="true"
                class="StyledIconBase-ea9ulj-0 lbJwfL sc-jqUVSM gKbZDu"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="StyledIconBase-ea9ulj-0 lbJwfL sc-jqUVSM gKbZDu"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="StyledIconBase-ea9ulj-0 lbJwfL sc-kDDrLX ghEnkw"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="StyledIconBase-ea9ulj-0 lbJwfL sc-kDDrLX ghEnkw"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="StyledIconBase-ea9ulj-0 lbJwfL sc-kDDrLX ghEnkw"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21.947 9.179a1.001 1.001 0 0 0-.868-.676l-5.701-.453-2.467-5.461a.998.998 0 0 0-1.822-.001L8.622 8.05l-5.701.453a1 1 0 0 0-.619 1.713l4.213 4.107-1.49 6.452a1 1 0 0 0 1.53 1.057L12 18.202l5.445 3.63a1.001 1.001 0 0 0 1.517-1.106l-1.829-6.4 4.536-4.082c.297-.268.406-.686.278-1.065z"
                />
              </svg>
              <span
                class="sc-evZas ccJzNt"
              >
                (01/02/2022)
              </span>
            </p>
            <div
              class="sc-iqcoie gAUmKf"
            >
              <p>
                Not so good product
              </p>
              <p
                class="sc-crXcEl gQMZmj"
              >
                Not recomended...
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      Trends
    </div>
  </section>
</div>
`;
